From 1c97069c9f360deae6996196ab7745d5c8c80d9c Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sat, 11 Aug 2018 14:48:52 +0300
Subject: [PATCH] hl: make: fix {dali,gaudi}-distclean

On new systems the build path is yet to be created.
Therefore the {dali,gaudi}-distclean fails with:
$ make dali-distclean
/bin/sh: 1: cd: can't cd to /home/dbarak/trees/npu-stack/buildroot-external/output/build
Makefile:162: recipe for target 'dali-distclean' failed
make: *** [dali-distclean] Error 2

Fix by checking the path exits, otherwise - nothing to do.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index 293b751..e8488d8 100644
--- a/Makefile
+++ b/Makefile
@@ -159,13 +159,15 @@
 
 .PHONY: dali-distclean gaudi-distclean
 dali-distclean gaudi-distclean:
-	@rm  -f $(HL_VARS_FILE)
-	@cd $(OUTPUT_DIR)/build && \
-	rm -rf \
-	uboot-custom linux-custom armcp-custom \
-	spdlog-custom specs-custom specs_external-custom \
-	host-uboot-test-hooks-custom host-tbot-custom \
-	thermal-custom preboot-custom host-fw_loader-custom
+	@rm -f $(HL_VARS_FILE)
+	@if [ -d $(OUTPUT_DIR)/build ]; then \
+		cd $(OUTPUT_DIR)/build && \
+		rm -rf \
+		uboot-custom linux-custom armcp-custom \
+		spdlog-custom specs-custom specs_external-custom \
+		host-uboot-test-hooks-custom host-tbot-custom \
+		thermal-custom preboot-custom host-fw_loader-custom; \
+	fi
 
 .PHONY: dali-onix-test dali-pldm-test dali-asic-test
 dali-onix-test: dali-onix-tools-install
